### PR TITLE
Recalculate intermediate hashes using current tree.

### PR DESCRIPTION
**Pull request by AlCutter**

### CODE REVIEW

File: merkle/merkle_path.go [hunk 1]
```diff
@@ -38,14 +38,35 @@ func (n NodeFetch) Equivalent(other NodeFetch) bool {
 	return n.Rehash == other.Rehash && n.NodeID.Equivalent(other.NodeID)
 }
 
+// checkSnapshot performs a couple of simple sanity checks on sa and treeSize
```
Comment (Martin2112) [inline merkle/merkle_path.go]: sa? typo?

File: merkle/merkle_path.go [hunk 2]
```diff
@@ -38,14 +38,35 @@ func (n NodeFetch) Equivalent(other NodeFetch) bool {
...
+// and returns an error if there's a problem.
+func checkSnapshot(ssDesc string, ss, treeSize int64) error {
+	if ss < 1 {
+		return fmt.Errorf("%s %d < 1", ssDesc, ss)
```
Comment (Martin2112) [inline merkle/merkle_path.go]: can you add something to the error message to show where it's from.

File: merkle/merkle_path.go (hunk 1 above)
Comment (AlCutter) [inline merkle/merkle_path.go]: whoops, fixed.

File: merkle/merkle_path.go (hunk 2 above)
Comment (AlCutter) [inline merkle/merkle_path.go]: Done at the next level up, how's that?

Comment (Martin2112) [inline merkle/merkle_path.go]: LGTM.

**Pull request merged**